sympy/series/sequences.py

```diff
@@ -0,0 +1,1144 @@
+    Only SymPy's :class:`Lambda` functions are permitted.
+
+    Examples
+    ========
+
+    >>> from sympy import SeqFunc, oo, Lambda, Symbol
+    >>> n = Symbol('n')
+    >>> s = SeqFunc(Lambda(n, n**2), (0, 5))
+    >>> s.function
+    Lambda(n, n**2)
+
+    For value at a particular point
+
+    >>> s.coeff(3)
+    9
+
+    supports slicing
+
+    >>> s[:]
+    [0, 1, 4, 9, 16, 25]
+
+    iterable
+
+    >>> list(s)
+    [0, 1, 4, 9, 16, 25]
+
+    changing step size
+
+    >>> SeqFunc(Lambda(n, n**2), (0, 5, 2))[:]
+    [0, 4, 16]
+
+    sequence starts from negative infinity
+
+    >>> SeqFunc(Lambda(n, n**2), (-oo, 0))[0:6]
+    [0, 1, 4, 9, 16, 25]
+
+    See Also
+    ========
+
+    sympy.series.sequences.SeqPer
+    sympy.series.sequences.SeqFormula
+    sympy.core.function.Lambda
+    """
+
+    is_SeqFunc = True
+
+    @staticmethod
+    def _validate(function):
+        if len(function.variables) != 1:
+            raise ValueError("Only single argument functions are allowed")
+        return sympify(function)
+
+    @property
+    def function(self):
+        return self.gen
+
+    @property
+    def variables(self):
+        return self.gen.variables
+
+    def _eval_coeff(self, pt):
+        return self.function(pt)
+
+    def _add(self, other):
+        """See docstring of SeqBase._add"""
+        if other.is_SeqFunc:
+            func1, v1 = self.function, self.variables[0]
+            func2, v2 = other.function, other.variables[0]
+            function = Lambda(v1, func1.expr + func2.expr.subs(v2, v1))
+            return SeqFunc(function, self._intersect_interval(other))
+
+    def _mul(self, other):
+        """See docstring of SeqBase._mul"""
+        if other.is_SeqFunc:
+            func1, v1 = self.function, self.variables[0]
+            func2, v2 = other.function, other.variables[0]
+            function = Lambda(v1, func1.expr * func2.expr.subs(v2, v1))
+            return SeqFunc(function, self._intersect_interval(other))
+
+    def coeff_mul(self, coeff):
+        """See docstring of SeqBase.coeff_mul"""
+        coeff = sympify(coeff)
+        expr = self.function.expr * coeff
+        return SeqFunc(Lambda(self.variables[0], expr), \
+                          (self.interval, self.step))
+
+
+def sequence(**kwargs):
+    """Returns appropriate sequence object.
+
+    Examples
+    ========
+
+    >>> from sympy import sequence, SeqPer, SeqFunc, SeqFormula, Lambda
+    >>> from sympy.abc import n
+
+    >>> sequence(formula=(n**2, n), interval=(0, 5))
+    SeqFormula((n**2, n), ([0, 5], 1))
+
+    >>> sequence(periodical=(1, 2, 3), interval=(0, 5))
+    SeqPer((1, 2, 3), ([0, 5], 1))
+
+    >>> sequence(func=Lambda(n, n**2), interval=(0, 5))
+    SeqFunc(Lambda(n, n**2), ([0, 5], 1))
+
+    See Also
+    ========
+
+    sympy.series.sequences.SeqPer
+    sympy.series.sequences.SeqFormula
+    sympy.series.sequences.SeqFunc
+    """
+    interval = kwargs.pop('interval', None)
+    if interval == None:
```

Comment:
Checking against None should be `x is None` and `x is not None` rather than `x == None` and `not x == None`